src/tests/system/tests/test_netgroups.py

```diff
@@ -108,3 +124,282 @@ def test_netgroups__add_remove_netgroup_member(client: Client, provider: Generic
+    ldap.ldap.modify(dev_users.dn, replace={"memberNisNetgroup": qa_users.dn})
+
+    client.sssd.dom("test")["entry_cache_timeout"] = "60"
+    client.sssd.start()
+
+    member = client.tools.getent.netgroup("DEVUsers").members
+    assert "(testhost5, ng9005, ldap.test)" in member
+    assert "(testhost1, ng9000, ldap.test)" in member
+
+
+@pytest.mark.importance("low")
+@pytest.mark.topology(KnownTopology.LDAP)
+def test_netgroup__different_syntax(client: Client, ldap: LDAP):
+    """
+    :title: Using different syntax for nisNetgroupTriple
+    :setup:
+        1. Create users, groups and start sssd.
+    :steps:
+        1. Check that the user ng9006 appears in the group members list, represented as the tuple (-,ng9006,).
+    :expectedresults:
+        1. The user ng9006 appears in the group members list
+    :customerscenario: False
+    """
+    ou = ldap.ou("Netgroup")
+    create_users(ldap)
+
+    qa_users = ldap.netgroup("QAUsers", basedn=ou).add()
+    qa_users.add_member(host="testhost1", user="ng9000", domain="ldap.test")
+
+    dev_users = ldap.netgroup("DEVUsers", basedn=ou).add()
+    dev_users.add_member(host="testhost5", user="ng9005", domain="ldap.test")
+    dev_users.add_member(user="ng9006")
+
+    client.sssd.dom("test")["entry_cache_timeout"] = "60"
+    client.sssd.start()
+
+    member = client.tools.getent.netgroup("DEVUsers").members
+    assert "(-,ng9006,)" in member
+
+
+@pytest.mark.importance("low")
+@pytest.mark.topology(KnownTopology.LDAP)
+def test_netgroup__host_and_domain(client: Client, ldap: LDAP):
+    """
+    :title: A scenario where an LDAP netgroup contains a member that
+        only has a host and domain specified, but no associated user.
+    :setup:
+        1.  Check that the tuple (samplehost, -, samplehost.domain.com) is part of the group
+    :expectedresults:
+        1. The tuple (samplehost, -, samplehost.domain.com) is part of the group
+    :customerscenario: False
+    """
+    ou = ldap.ou("Netgroup")
+    create_users(ldap)
+
+    qa_users = ldap.netgroup("QAUsers", basedn=ou).add()
+    qa_users.add_member(host="testhost1", user="ng9000", domain="ldap.test")
+
+    dev_users = ldap.netgroup("DEVUsers", basedn=ou).add()
+    dev_users.add_member(host="testhost5", user="ng9005", domain="ldap.test")
+    dev_users.add_member(host="samplehost", domain="samplehost.domain.com")
+
+    client.sssd.dom("test")["entry_cache_timeout"] = "60"
+    client.sssd.start()
+
+    member = client.tools.getent.netgroup("DEVUsers").members
+    assert "(samplehost,-,samplehost.domain.com)" in member
+
+
+@pytest.mark.importance("low")
+@pytest.mark.topology(KnownTopology.LDAP)
+def test_netgroup__with_nested_loop(client: Client, ldap: LDAP):
+    """
+    :title: Create and manages nested LDAP netgroups and tests their behavior
+        through several scenarios involving caching, membership queries, and restarts of the SSSD service.
```

Comment:
Odd line break.

Reply:
This script is formatted with black, isort, flake8, mypy. If change it it,  formation will break